File: binding/src/libqbindingstorage.cpp
#include <QBindingStatus>
#include <QBindingStorage>
#include <QUntypedPropertyData>
#include <qbindingstorage.h>
#include "libqbindingstorage.h"
#include "libqbindingstorage.hxx"

QBindingStatus* QBindingStatus_new(const QBindingStatus* other) {
	 return new QBindingStatus(*other);
}

QBindingStatus* QBindingStatus_new2(QBindingStatus* other) {
	 return new QBindingStatus(*other);
}

void QBindingStatus_CopyAssign(QBindingStatus* self, QBindingStatus* other) {
    *self = *other;
}

void QBindingStatus_MoveAssign(QBindingStatus* self, QBindingStatus* other) {
    *self = std::move(*other);
}

QtPrivate::BindingEvaluationState* QBindingStatus_CurrentlyEvaluatingBinding(const QBindingStatus* self) {
	return self->currentlyEvaluatingBinding;
}

void QBindingStatus_SetCurrentlyEvaluatingBinding(QBindingStatus* self, QtPrivate::BindingEvaluationState* currentlyEvaluatingBinding) {
	self->currentlyEvaluatingBinding;
}

QtPrivate::CompatPropertySafePoint* QBindingStatus_CurrentCompatProperty(const QBindingStatus* self) {
	return self->currentCompatProperty;
}

void QBindingStatus_SetCurrentCompatProperty(QBindingStatus* self, QtPrivate::CompatPropertySafePoint* currentCompatProperty) {
	self->currentCompatProperty;
}

QPropertyDelayedNotifications* QBindingStatus_GroupUpdateData(const QBindingStatus* self) {
	return self->groupUpdateData;
}

void QBindingStatus_SetGroupUpdateData(QBindingStatus* self, QPropertyDelayedNotifications* groupUpdateData) {
	self->groupUpdateData;
}

void QBindingStatus_Delete(QBindingStatus* self) {
    delete self;
}

QBindingStorage* QBindingStorage_new() {
	 return new QBindingStorage();
}

QBindingStorage* QBindingStorage_new2(const QBindingStorage* param1) {
	 return new QBindingStorage(*param1);
}

bool QBindingStorage_IsEmpty(QBindingStorage* self) {
	return self->isEmpty();
}

bool QBindingStorage_IsValid(const QBindingStorage* self) {
	return self->isValid();
}

const QBindingStatus* QBindingStorage_Status(const QBindingStorage* self, QtPrivate::QBindingStatusAccessToken param1) {
	return self->status(param1);
}

void QBindingStorage_RegisterDependency(const QBindingStorage* self, const QUntypedPropertyData* data) {
	self->registerDependency(data);
}

QtPrivate__QPropertyBindingData* QBindingStorage_BindingData(const QBindingStorage* self, const QUntypedPropertyData* data) {
	return self->bindingData(data);
}

QtPrivate__QPropertyBindingData* QBindingStorage_BindingData2(QBindingStorage* self, QUntypedPropertyData* data, bool create) {
	return self->bindingData(data, create);
}

void QBindingStorage_Delete(QBindingStorage* self) {
    delete self;
}

File: binding/src/libqbindingstorage.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qbindingstatus.html
pub const qbindingstatus = struct {

    /// New constructs a new QBindingStatus object.
    pub fn New(other: ?*anyopaque) QtC.QBindingStatus {
        return qtc.QBindingStatus_new(@ptrCast(other));
    }


    /// New2 constructs a new QBindingStatus object.
    pub fn New2(other: ?*anyopaque) QtC.QBindingStatus {
        return qtc.QBindingStatus_new2(@ptrCast(other));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QBindingStatus_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QBindingStatus_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn CurrentlyEvaluatingBinding(self: ?*anyopaque, ) QtPrivate::BindingEvaluationState {
        return @ptrCast(qtc.QBindingStatus_CurrentlyEvaluatingBinding(@ptrCast(self)));
    }

    pub fn SetCurrentlyEvaluatingBinding(self: ?*anyopaque, currentlyEvaluatingBinding: *QtPrivate::BindingEvaluationState) void {
        qtc.QBindingStatus_SetCurrentlyEvaluatingBinding(@ptrCast(self), @ptrCast(currentlyEvaluatingBinding));
    }

    pub fn CurrentCompatProperty(self: ?*anyopaque, ) QtPrivate::CompatPropertySafePoint {
        return @ptrCast(qtc.QBindingStatus_CurrentCompatProperty(@ptrCast(self)));
    }

    pub fn SetCurrentCompatProperty(self: ?*anyopaque, currentCompatProperty: *QtPrivate::CompatPropertySafePoint) void {
        qtc.QBindingStatus_SetCurrentCompatProperty(@ptrCast(self), @ptrCast(currentCompatProperty));
    }

    pub fn GroupUpdateData(self: ?*anyopaque, ) QPropertyDelayedNotifications {
        return @ptrCast(qtc.QBindingStatus_GroupUpdateData(@ptrCast(self)));
    }

    pub fn SetGroupUpdateData(self: ?*anyopaque, groupUpdateData: *QPropertyDelayedNotifications) void {
        qtc.QBindingStatus_SetGroupUpdateData(@ptrCast(self), @ptrCast(groupUpdateData));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QBindingStatus_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qbindingstorage.html
pub const qbindingstorage = struct {

    /// New constructs a new QBindingStorage object.
    pub fn New() QtC.QBindingStorage {
        return qtc.QBindingStorage_new();
    }


    /// New2 constructs a new QBindingStorage object.
    pub fn New2(param1: ?*anyopaque) QtC.QBindingStorage {
        return qtc.QBindingStorage_new2(@ptrCast(param1));
    }


    pub fn IsEmpty(self: ?*anyopaque, ) bool {
        return qtc.QBindingStorage_IsEmpty(@ptrCast(self));
    }

    pub fn IsValid(self: ?*anyopaque, ) bool {
        return qtc.QBindingStorage_IsValid(@ptrCast(self));
    }

    pub fn Status(self: ?*anyopaque, param1: QtPrivate::QBindingStatusAccessToken) QtC.QBindingStatus {
        return qtc.QBindingStorage_Status(@ptrCast(self), param1);
    }

    pub fn RegisterDependency(self: ?*anyopaque, data: ?*anyopaque) void {
        qtc.QBindingStorage_RegisterDependency(@ptrCast(self), @ptrCast(data));
    }

    pub fn BindingData(self: ?*anyopaque, data: ?*anyopaque) QtC.QtPrivate__QPropertyBindingData {
        return qtc.QBindingStorage_BindingData(@ptrCast(self), @ptrCast(data));
    }

    pub fn BindingData2(self: ?*anyopaque, data: ?*anyopaque, create: bool) QtC.QtPrivate__QPropertyBindingData {
        return qtc.QBindingStorage_BindingData2(@ptrCast(self), @ptrCast(data), create);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QBindingStorage_Delete(@ptrCast(self));
    }
};
